main.go

```diff
@@ -16,34 +16,37 @@ package main
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
 	"net"
+	"net/http"
 	"os"
-	"reflect"
-	"sort"
 	"strings"
 	"time"
 
-	"github.com/scaleway/prometheus-scw-sd/model"
-
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
-	api "github.com/scaleway/go-scaleway"
-	scw "github.com/scaleway/go-scaleway/types"
-	"github.com/scaleway/prometheus-scw-sd/adapter"
-	"github.com/scaleway/prometheus-scw-sd/targetgroup"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/scaleway/go-scaleway"
+	"github.com/scaleway/go-scaleway/types"
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
 var (
-	a          = kingpin.New("sd adapter usage", "Tool to generate file_sd target files for unimplemented SD mechanisms.")
-	token      = a.Flag("token", "The token for Scaleway API.").Required().String()
-	private    = a.Flag("private", "Use servers private IP.").Bool()
-	outputFile = a.Flag("output.file", "Output file for file_sd compatible file.").Default("scw_sd.json").String()
-	port       = a.Flag("port", "Port on which to scrape metrics.").Default("9100").Int()
-	interval   = a.Flag("time.interval", "Time in second to wait between each refresh.").Default("90").Int()
-	logger     log.Logger
-
-	scwPrefix = model.MetaLabelPrefix + "scw_"
+	a            = kingpin.New("sd adapter usage", "Tool to generate Prometheus file_sd target files for Scaleway.")
+	outputf      = a.Flag("output.file", "The output filename for file_sd compatible file.").Default("scw.json").String()
+	organization = a.Flag("scw.organization", "The Scaleway organization.").Default("").String()
+	region       = a.Flag("scw.region", "The Scaleway region.").Default("").String()
+	token        = a.Flag("scw.token", "The authentication token (secret key).").Default("").String()
+	tokenf       = a.Flag("scw.token-file", "The authentication token file.").Default("").String()
+	refresh      = a.Flag("target.refresh", "The refresh interval (in seconds).").Default("30").Int()
+	port         = a.Flag("target.port", "The default port number for targets.").Default("80").Int()
+	listen       = a.Flag("web.listen-address", "The listen address.").Default(":9465").String()
```

Comment:
As 9465 had been reserved in between, I've just registered 9361 on https://github.com/prometheus/prometheus/wiki/Default-port-allocations